File: csharp-ms/SupermarketReceipt/Discount.cs
namespace SupermarketReceipt
{
    public class Discount
    {
        public Discount(Product product, string description, double discountAmount)
        {
            Product = product;
            Description = description;
            DiscountAmount = discountAmount;
        }

        public string Description { get; }
        public double DiscountAmount { get; }
        public Product Product { get; }
    }
}

File: dart/lib/supermarket/model/discount.dart
import 'package:supermarket_receipt/supermarket/model/product.dart';

class Discount {

  final Product _product;
  final String _description;
  final double _discountAmount;

  Discount(this._product, this._description, this._discountAmount);

  String getDescription() => _description;

  double getDiscountAmount() => _discountAmount;

  Product getProduct() => _product;
}